src/vs/base/browser/ui/hover/hover.ts

```diff
@@ -0,0 +1,269 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import type { IHoverDelegate } from 'vs/base/browser/ui/hover/hoverDelegate';
+import type { HoverPosition } from 'vs/base/browser/ui/hover/hoverWidget';
+import type { CancellationToken } from 'vs/base/common/cancellation';
+import type { IMarkdownString } from 'vs/base/common/htmlContent';
+import type { IDisposable } from 'vs/base/common/lifecycle';
+
+/**
+ * Enables the convenient display of rich markdown-based hovers in the workbench.
+ */
+export interface IHoverDelegate2 {
+	/**
+	 * Shows a hover, provided a hover with the same options object is not already visible.
+	 * @param options A set of options defining the characteristics of the hover.
+	 * @param focus Whether to focus the hover (useful for keyboard accessibility).
+	 *
+	 * **Example:** A simple usage with a single element target.
+	 *
+	 * ```typescript
+	 * showHover({
+	 *   text: new MarkdownString('Hello world'),
+	 *   target: someElement
+	 * });
+	 * ```
+	 */
+	showHover(options: IHoverOptions, focus?: boolean): IHoverWidget | undefined;
+
+	/**
+	 * Hides the hover if it was visible. This call will be ignored if the the hover is currently
+	 * "locked" via the alt/option key.
+	 */
+	hideHover(): void;
+
+	/**
+	 * This should only be used until we have the ability to show multiple context views
+	 * simultaneously. #188822
+	 */
+	showAndFocusLastHover(): void;
+
+	// TODO: Change hoverDelegate arg to exclude the actual delegate and instead use the new options
+	setupUpdatableHover(hoverDelegate: IHoverDelegate, htmlElement: HTMLElement, content: IUpdatableHoverContentOrFactory, options?: IUpdatableHoverOptions): IUpdatableHover;
```

Comment:
I don't like the name of this method. It does not indicate to the consumer of the hover service that this ties a hover to an HTMLElement.

Reply:
@benibenj what would you suggest? Also doesn't `showHover` also ties it to an element via the hover target?

Reply:
We'll fix this in a follow up